hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -1430,6 +1431,17 @@ static String getRemoteClientIpAndPort() {
     return Address.fromParts(address.getHostAddress(), rpcCall.getRemotePort()).toString();
   }
 
+  /**
+   * @return Remote client's username.
+   */
+  static String getUserName() {
```

Comment:
Why not return optional here?

Reply:
> Why not return optional here?

Thanks @Apache9 for your review. Because the final result is turned into a String, in keeping with the style of this method (getRemoteLientiPandPort()).

Reply:
Oh, it is only called inside RSRpcService? So it should be private, but for testing, we make it package private. Then please add a RestrictApi annotation here.